core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -37,42 +37,162 @@
   /** A map from default property key's string name to the key. */
   private static final Map<PropertyKey, Object> DEFAULT_VALUES = new HashMap<>();
 
+  /**
+   * Builder to create {@link PropertyKey} instances.
+   */
+  public static final class Builder {
+    private String[] mAlias;
+    private Object mDefaultValue;
+    private String mDescription;
+    private final String mName;
+
+    /**
+     * @param name name of this property to build.
+     */
+    public Builder(String name) {
+      mName = name;
+    }
+
+    /**
+     * @param template template of the name of the property to build
+     * @param params parameters of the template
+     */
+    public Builder(PropertyKey.Template template, Object... params) {
+      mName = String.format(template.mFormat, params);
+    }
+
+    /**
+     * @param alias alias of this property key to build.
+     * @return the updated builder instance
+     */
+    public Builder setAlias(String[] alias) {
+      mAlias = alias;
+      return this;
+    }
+
+    /**
+     * @param defaultValue default value of this property key to build.
+     * @return the updated builder instance
+     */
+    public Builder setDefaultValue(Object defaultValue) {
+      mDefaultValue = defaultValue;
+      return this;
+    }
+
+    /**
+     * @param description of this property key to build.
+     * @return the updated builder instance
+     */
+    public Builder setDescription(String description) {
+      mDescription = description;
+      return this;
+    }
+
+    /**
+     * @return the created property key instance
+     */
+    public PropertyKey build() {
+      return PropertyKey.create(mName, mDefaultValue, mAlias);
```

Comment:
wonder why we don't  put mDescription here. Or this is pending work after another two jira on this class?